src/parse.js

```diff
@@ -36,136 +35,68 @@ function parseData ({data, parseFn} = {}) {
 }
 
 /**
  * Parse patterns files and build data object.
- *
- * @param {Object} options with:
- *   {glob} patterns   Where to look for patterns files
- *   {patternKey}      String key for patterns directory, naming
- * @TODO I still don't like the coupling between patternKey and directories
- * @return {Object} Fully-built patterns data object
+ * @TODO document
  */
-function parsePatterns ({patterns, patternKey} = {}) {
-  const patternData = {};
-  return utils.readFiles(patterns, {
-    contentFn: (contents, path) => frontMatter(contents)
-  }).then(fileData => {
-    fileData.forEach(patternFile => {
-      const keys = utils.relativePathArray(patternFile.path, patternKey);
-      const entryKey = utils.keyname(patternFile.path, { stripNumbers: false });
-      const pathKey = utils.keyname(patternFile.path);
-      getPatternEntry(keys, patternData)[entryKey] = {
-        name: utils.titleCase(pathKey),
-        id  : keys.concat(pathKey).join('.'),
-        data: patternFile.contents.attributes,
-        contents: patternFile.contents.body
-      };
-    });
-    // @TODO Figure out how to emulate "local namespacing" of HBS vars
-    // so that one can reference data from front matter in patterns
-    // @TODO Run some fields through markdown
-    // @TODO Do we need to trim whitespace from pattern content?
-    // @TODO Do we need to store pattern data on another object as well?
-    // @TODO Do we need any further sorting?
-    // @TODO Need to register Handlebars partial
-    return patternData;
-  });
+function parsePatterns (patterns, options) {
+  return parseRecursive(patterns, options.keys.patterns, options);
+
+  // @TODO Figure out how to emulate "local namespacing" of HBS vars
+  // so that one can reference data from front matter in patterns
+  // @TODO Run some fields through markdown
+  // @TODO Do we need to trim whitespace from pattern content?
+  // @TODO Do we need to store pattern data on another object as well?
+  // @TODO Do we need any further sorting?
+  // @TODO Need to register Handlebars partial
 }
 
 function parseAll (options = {}) {
   return Promise.all([
-    parseData({
-      data: options.data,
-      parseFn: options.dataFn
-    }),
-    parseDocs({
-      docs: options.docs,
-      parseFn: options.docsFn
-    }),
-    parseLayouts(options),
-    parsePages({ pages: options.pages }),
-    parsePatterns({
-      patterns: options.patterns,
-      patternKey: options.keys.patterns
-    })
+    parseData(options.data, options),
+    parseDocs(options.docs, options), // TODO No such thing as docs
+    parseLayouts(options.layouts, options),
+    parsePages(options.pages, options),
+    parsePatterns(options.patterns, options)
   ]).then(allData => {
     return {
       data    : allData[0],
-      docs    : allData[1],
-      layouts : allData[2],
+      docs    : allData[1], // TODO no such thing
+      layouts : allData[2], // TODO Does this really belong here?
```

Comment:
Maybe we can use array destructuring here instead of referencing the indices? 

Probably too clever.

```js
.then(([data, docs, layouts, pages, patterns]) => Object.assign({}, { data, docs, layouts, ...))
```